src/recovery/cubic.rs

```diff
@@ -213,13 +213,12 @@ fn on_packet_acked(
     // a few packets (less than RESTORE_COUNT_THRESHOLD), it's considered
     // as spurious and restore to the previous state.
     if r.congestion_recovery_start_time.is_some() {
-        if let Some(prior) = &r.cubic_state.prior {
-            let new_lost = r.lost_count - prior.lost_count;
+        let new_lost = r.lost_count - r.cubic_state.prior.lost_count;
 
-            if r.congestion_window < prior.congestion_window &&
-                new_lost < RESTORE_COUNT_THRESHOLD
-            {
-                rollback(r);
+        if new_lost < RESTORE_COUNT_THRESHOLD {
+            let did_rollback = rollback(r);
+
+            if did_rollback {
                 return;
```

Comment:
```suggestion
            if rollback(r) {
                return;
```

Reply:
So, I did that originally, but then clippy complained that the `if`s could be merged together, though I found:
```rust
if new_lost < RESTORE_COUNT_THRESHOLD && rollback() {
```
kind of hard to read (as the `rollback()` call looks a bit hidden, and would be easy to pass over when reading), so I did it this way to silence clippy.

Reply:
no problem then. 👍